### ISSUE TITLE
Ventura: windows not getting "top window" when got focus with Cmd-Tab

### ISSUE THREAD

**Issue by hwde** [XQuartz/XQuartz#293]
Since the upgrade to macOS 13 (Ventura) the windows of XQuartz (for example xterm) stay in the background when switching the active application with Cmd-Tab. Is this something that can be improved by setting or configuration? The windows still get the focus ... only it is not the "top window"? Does no one else have this problem?

How to reproduce the problem?

- Start any program (e.g. Chrome Browser)
- start new "xterm
- Cmd-Tab (Chrome has the focus and is "top window")
- work with Chrome
- Cmd-Tab (xterm has the focus, but Chrome is still "top window

**Comment by dpeterc** [comment]
I am also having this problem on Ventura.
My Motif based software uses many windows in parallel. There is top level application and many dialog windows for various purposes. Some of these windows have their own child window. Often, when a window is closed, the focus is passed to the parent window, but the window does not fully get to the front. So user is clicking on buttons of the top window, but mouse click is passed to the window in focus, which is in the back.
The only way to bring the top window in focus is to click on it on area, which does not overlap the focus window in the back. Or to select "Window > Bring All to Front", from XQuartz menu.
Sometimes, the whole application disappears, due to this wrong focus passing. But is it not crashed, just "iconized" in invisible way. It can be brought back by clicking "Window > Bring All to Front", from XQuartz menu.
Of course, such way of work is very uncomfortable.

I attach a picture which illustrates the problem of a window in focus which is overlapped with another window.
Window in focus should always be in front, such is normal X11 behavior and it has worked in all Quartz versions before Ventura.

![image](https://user-images.githubusercontent.com/15419249/202764290-7cc094f5-fa1b-49bd-8933-7f9ce95bc91a.png)

**Comment by eraineri** [comment]
Same issue, I noticed that if you cmd+tab back one more time it does raise the window.  But you need to do it 2 times.

**Comment by arsenhakobyan** [comment]
Same here. Please rise the priority of this issue.

**Comment by jeremyhu** [comment]
Please file a report via http://bugreport.apple.com and report back the FB# here for tracking.

**Comment by sirainen** [comment]
I've never before used feedbackassistant, but this link just gives "Feedback Not Found". Is that intentional? Or is it just private and could have been marked as public?

Anyway, this is rather annoying bug and slows down my work..

**Comment by hwde** [comment]
I never used it before, "feedback" seems to be private.

**Comment by jeremyhu** [comment]
It is private, but the link gave me what I needed (the number) to followup internally.

**Comment by dpeterc** [comment]
I have just installed Ventura 13.1 and the problem has not been solved with the new MacOS release.

**Comment by BethLong06** [comment]
I updated to XQuartz 2.8.4 and I'm still having this problem. I've submitted a bug report, number FB11907433. Hope this will be fixed soon!

**Comment by jeremyhu** [comment]
Yep, it is already tracked with @hwde's report above.  I'll comment here when a fix is available for testing / verification.

**Comment by almostjulian** [comment]
Unchecking `Focus On New Windows` seems to allow tab switching, obviously at the cost not having new windows move to the foreground.

**Comment by hwde** [comment]
> Unchecking `Focus On New Windows` seems to allow tab switching, obviously at the cost not having new windows move to the foreground.

Too bad, unchecking that option doesn't change anything here.

**Comment by sirainen** [comment]
It didn't first work with me either, but then I upgraded to 2.8.4 and now it works with `Focus On New Windows` disabled.

**Comment by sirainen** [comment]
> It didn't first work with me either, but then I upgraded to 2.8.4 and now it works with Focus On New Windows disabled.

..and then it broke again after a while.

**Comment by dpeterc** [comment]
I have tried it again on Ventura 13.2 and XQuartz-2.8.5 and the problem persists.
I wish Apple would give more complete release notes for its OS updates, it is hard to guess from these three bullets if our issues will ever be resolved.
https://support.apple.com/en-us/HT213268

**Comment by jeremyhu** [comment]
Please verify this as fixed in the macOS 13.3 beta that was just released.

**Comment by dpeterc** [comment]
I have installed the latest Ventura 13.3 Beta (22E5219e) on iMac 27" Intel i9, with latest XQuartz 2.8.5, and I am sorry to report that the problem has not been solved.

**Comment by jeremyhu** [comment]
I just tried, and it worked for me.  Can you file a new bug report with a video (and sysdiagnose) of the behavior you are seeing with the current macOS beta?

**Comment by skchronicles** [comment]
@jeremyhu Can you confirm that this issue has been resolved? Did the upgrade to macOS 13.3 fix the issue? I just see conflicting information in the last two comments, and you marked the issue as complete. I am just curious because I am running into the same issue with a python tkinter application that is running over an X11 connection with XQuartz. It is worth noting that I am not running into the issue with macOS Monterey; however, the problem persists with Ventura.

**Comment by jeremyhu** [comment]
Yes, this issue is resolved.  Based on @dpeterc 's comment in another ticket, I think their issue is something else.

**Comment by dpeterc** [comment]
I have posted a new ticket, so it is easier to track if the problem is something else:
https://github.com/XQuartz/XQuartz/issues/328

**Comment by hwde** [comment]
> `Focus On New Windows` disabled and then enabled, and it works - for now at least. Ventura 13.2.1 and Quartz 2.8.5

... first it seems to work ... but few minutes later it doesn't. Press thumbs that 13.3 will fix it.

**Comment by karog** [comment]
I just upgraded to 13.3 on 3 different macs (mac mini m2 pro, macbook air m2, 2017 macbook) and the problem is fixed on all of them. Cmd-Tab brings all XQuartz windows forward for me.

**Comment by arsenhakobyan** [comment]
Finally. Many thanks, to the person who fixed this! I was waiting for it in each release.
